tests/test_activation_responses.py

```diff
@@ -0,0 +1,277 @@
+    rows[6] = (0, 'n-16', '12 h', 0.0000e0)
+    rows[7] = (0, 'n-16', '3 d', 0.0000e0)
+    rows[8] = (0, 'TOTAL', 'shutdown', 3.1588e-9)
+    rows[9] = (0, 'TOTAL', '1000 s', 9.5258e-18)
+    rows[10] = (0, 'TOTAL', '12 h', 9.5251e-18)
+    rows[11] = (0, 'TOTAL', '3 d', 9.5214e-18)
+    tab.append(rows[:])
+    # close the file
+    h5f.close()
+    f.close()
+
+
+def test_response_to_hdf5():
+    """
+    This function test alara.response_to_hdf5, with response of:
+        - decay_heat
+        - specific_activity
+        - alpha_heat
+        - beta_heat
+        - gamma_heat
+        - wdr
+        - photon_source
+    """
+    # skip test if h5diff not exist
+    is_h5diff = os.system('which h5diff')
+    if is_h5diff != 0:
+        raise SkipTest
+
+    for response in responses:
+        # read  output.txt and write h5 file
+        filename = os.path.join(thisdir, "files_test_activation_responses", ''.join([response, '_output.txt']))
+        h5_filename = os.path.join(thisdir, "files_test_activation_responses", ''.join([response, '.h5']))
+        response_to_hdf5(filename, response)
+
+        # generate expected h5 file
+        exp_h5_filename = os.path.join(thisdir, "files_test_activation_responses", ''.join(['exp_', response, '.h5']))
+        _generate_exp_h5(filename, response, exp_h5_filename)
+
+        # compare two h5 files
+        command = ''.join(['h5diff ', h5_filename, ' ', exp_h5_filename])
+        diff_flag = os.system(command)
+        # return value 0 if no difference, 1 if differences found, 2 if error
+        assert_equal(diff_flag, 0)
+
+        # remove generated files
+        os.remove(h5_filename)
+        os.remove(exp_h5_filename)
+
+
+def test_responses_to_hdf5_multiple():
+    """
+    This function test alara.response_to_hdf5, read output.txt of multiple responses:
+        - decay_heat
+        - specific_activity
+        - alpha_heat
+        - beta_heat
+        - gamma_heat
+        - wdr
+        - photon_source
+    """
+    # skip test if h5diff not exist
+    is_h5diff = os.system('which h5diff')
+    if is_h5diff != 0:
+        raise SkipTest
+
+    for response in responses:
+        # read  output.txt and write h5 file
+        filename = os.path.join(thisdir, "files_test_activation_responses", 'multiple_output.txt')
+        h5_filename = os.path.join(thisdir, "files_test_activation_responses", ''.join([response, '.h5']))
+        response_to_hdf5(filename, response)
+
+        # generate expected h5 file
+        exp_h5_filename = os.path.join(thisdir, "files_test_activation_responses", ''.join(['exp_', response, '.h5']))
+        _generate_exp_h5(filename, response, exp_h5_filename)
+
+        # compare two h5 files
+        command = ''.join(['h5diff ', h5_filename, ' ', exp_h5_filename])
+        diff_flag = os.system(command)
+        # return value 0 if no difference, 1 if differences found, 2 if error
+        assert_equal(diff_flag, 0)
```

Comment:
If this test fails, it's not obvious to me that the user will learn what is different between the two h5 files in the test output. Will the h5diff output be printed? I think the failure case will simply say something like "diff_flag=2, expected 0."

Reply:
Well, the comparison of the `h5` files is actually put here for developers, but not for users. Users who install the pyne will not encounter a failure of the test.
This `h5diff` output is not printed.

Once the error happens, it usually means the data in the `h5` file are different, the developer should check the data (with `h5dum`) and find out the reason.